### PR TITLE
GH-7787: Migrate file_attachment to PureComponent

### PR DESCRIPTION
**Pull request by jespino**
#### Ticket Link
mattermost/mattermost-server#7787

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: tests/components/file_attachment.test.jsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import {shallow, mount} from 'enzyme';
+
+import FileAttachment from 'components/file_attachment.jsx';
+
+jest.mock('utils/utils.jsx', () => {
+    const original = require.requireActual('utils/utils.jsx');
+    return {
+        ...original,
+        loadImage: jest.fn((id, callback) => callback())
+    };
```
Comment (saturninoabril) [inline tests/components/file_attachment.test.jsx]: I think `canDownloadFiles` should have been included here. Doing so will render correct snapshot for `without compact display and without can download`.
```
        loadImage: jest.fn((id, callback) => callback()),
        canDownloadFiles: jest.fn(() => true)
```

File: tests/components/file_attachment.test.jsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+});
+
+jest.mock('utils/file_utils', () => {
```
Comment (saturninoabril) [inline tests/components/file_attachment.test.jsx]: This can be removed.

File: tests/components/file_attachment.test.jsx (hunk 1 above)
Comment (jespino) [reply tests/components/file_attachment.test.jsx]: canDownloadFiles is from utils/file_utils and loadImage is from utils/utils.

File: tests/components/file_attachment.test.jsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            fileInfo={fileInfoProp}
+            handleImageClick={handleImageClickProp}
+            index={indexProp}
+            compactDisplay={compactDisplay}
+        />
+    );
+}
+
+describe('component/FileAttachment', () => {
+    afterAll(() => {
```
[the remark this answers is not in the document]
Comment (jespino) [reply tests/components/file_attachment.test.jsx]: Fixed! :)

File: tests/components/file_attachment.test.jsx (hunk 1 above)
Comment (saturninoabril) [reply tests/components/file_attachment.test.jsx]: I see, you're correct. Thanks for clarifying!

**Pull request merged**